client/src/containers/InsertLinkModal/InsertLinkModal.js

```diff
@@ -1,68 +1,71 @@
-import React, { Component } from 'react';
+import React, { useEffect, useRef } from 'react';
 import PropTypes from 'prop-types';
 import { bindActionCreators, compose } from 'redux';
 import { connect } from 'react-redux';
 import FormBuilderModal from 'components/FormBuilderModal/FormBuilderModal';
 import fileSchemaModalHandler from 'containers/InsertLinkModal/fileSchemaModalHandler';
 import * as schemaActions from 'state/schema/SchemaActions';
 
-class InsertLinkModal extends Component {
-  constructor(props) {
-    super(props);
+const InsertLinkModal = (props) => {
+  const {
+    isOpen,
+    onInsert,
+    onClosed,
+    setOverrides,
+  } = props;
 
-    this.handleSubmit = this.handleSubmit.bind(this);
-    if (!props.isOpen) {
-      props.setOverrides(null);
+  const prevIsOpenRef = useRef(isOpen);
+
+  useEffect(() => {
+    if (!isOpen) {
+      setOverrides(null);
     }
-  }
+  }, []);
 
-  componentDidUpdate(oldProps) {
-    const props = this.props;
-    if ((props.isOpen && !oldProps.isOpen) || (!props.isOpen && oldProps.isOpen)) {
-      props.setOverrides(props.isOpen ? props : null);
+  useEffect(() => {
+    const prevIsOpen = prevIsOpenRef.current;
+    if ((isOpen && !prevIsOpen) || (!isOpen && prevIsOpen)) {
+      setOverrides(isOpen ? props : null);
     }
-  }
+    prevIsOpenRef.current = isOpen;
+  }, [isOpen]);
```

Comment:
Since `isOpen` is a bool, isn't this just equivalent to the below?

```js
useEffect(() => {
  setOverrides(isOpen ? props : null);
}, [isOpen]);
```